Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM ubuntu:latest
+
+# Install the dependencies. We'll use the ubuntu provided mlton to bootstrap our local build.
+RUN apt-get update -y -qq \
+ && apt-get install -y -qq git build-essential libgmp-dev libtool automake mlton mlton-tools 
+
+# Copy the current directory (MLton source root) to a location within the container & move there
+COPY . /root/mlton
+WORKDIR /root/mlton
+
+# Build from source & install
+RUN autoreconf -vfi \
+ && ./configure \
+ && make \
+ && make install \
+ && rsync -a /usr/local/usr/ /usr/local/
```

Comment:
I want to call attention to this last line. I don't like it, but with the current Makefile it seems unavoidable that the install location must end with "/usr" on linux. 

Looking at the issues it seems it's already in scope of issue #131.

Reply:
@rcythr Perhaps pass a `--prefix` option to the `./configure` step to install under an arbitrary path and symlink?

Reply:
The ubuntu package **mlton** gets installed into `/usr/bin`, so I wanted to install to `/usr/local/bin` since it comes earlier on Ubuntu's `$PATH`.

If I understand `Makefile.in` correctly, setting `--prefix` on `./configure` will set the `$DESTDIR` variable, but `$PREFIX` will remain unchanged. https://github.com/MLton/mlton/blob/master/Makefile.in#L315

Since `$TBIN` and `$TLIB` will be set to `$(DESTDIR)$(prefix)` my `$TLIB` and `$TBIN` paths will necessarily end with `/usr`. In this case `$DESTDIR` is `/usr/local`, so it will be `/usr/local/usr`. Alternatively I could set `--prefix` to "", which would put the install in `/usr`.

The following definitely works, and is better than my current approach using rsync:

    make install PREFIX=

I'm currently rebuilding and testing with this modification.

Reply:
Not honoring a `--prefix` passed to `./configure` seems like a problem.  Certainly not the intention with introducing a `./configure` (and I'm beginning to wonder if we shouldn't revert this partial switch to autotools, since there is really much much more that should be done to really support/use autotools).

Reply:
The build through autotools mostly seemed pretty seamless overall. I'm certainly more familiar with CMake, but that's a matter of preference and not a problem with autotools.

I think the only part that needs revision to fix this issue is the `Makefile.in` lines `315` through `333`. I think the fix is to simply remove the `$PREFIX` and `$prefix` makefile variables completely. Once that's done the `--prefix` from `./configure` will be assigned to `$DESTDIR`.

I'll try it and see if it works fine after that change. If it seems to work OK I'll create another branch & pull request with that change separate from this Dockerfile PR.

Reply:
@rcythr That's correct. Actually just did a PR on your branch with the changes

Reply:
@myegorov Looks good now. Build & regression are working as expected.

The addition of a wrapper script & mention in the README is a nice touch.

Reply:
@myegorov Actually, looking at run-docker more closely I do have one concern:

   docker rm $(docker ps -qa)
   docker rmi -f $(docker images -qa -f 'dangling=true')

These are very destructive commands to run against docker. They're essentially wiping out the users containers and images, which may be more than just MLton!

Reply:
@rcythr valid concern. Perhaps changing to this would be more reasonable?

```shell
NAME="mlton"

# remove all stopped containers (this is general housekeeping, might be omitted)
# equivalent to but more readable than `docker rm $(docker ps -qa)`
docker container prune

# remove old MLton images
docker images | grep $NAME | awk '{print $3}' | xargs docker rmi
```

Reply:
@myegorov

I pushed a couple of commits that should do essentially what you were intending, but slightly differently:

1. By doing an `docker image rm` based on an unlikely tag name `mlton-run-docker`, we make it unlikely a user will have an image accidentally deleted. If we want to be really careful about this we could probably generate a UUID or something instead.

2. By using `docker run --rm` we tell docker to automatically delete the container once we're done with it. This is preferred over pruning because it is *possible* that a user has a stopped container which they plan to restart in the future.